syntax = "proto2";
package MSG;

// message type
enum MsgId {
	CLOSED = 10000;

	LOGIN_REQ = 10101;
	LOGIN_ANS = 10102;

	PING_REQ = 10103;
	PONG_ANS = 10104;

	REGIST_REQ = 10105;
	REGIST_ANS = 10106;


}

// Error type
enum ErrorCode {
	ERR_SUCCESS = 0;
	ERR_LOGIN_FAIL = 1;
	ERR_ARGUMENT_FAIL = 2;
	ERR_AUTHORITY_FAIL = 3;
	ERR_SESSIONKEY_FAIL = 4;
}

// platform type
enum PlatForm {
	IOS = 0;
	ANDROID = 1;
}


// user data
message UserData_ {

	required uint64 uid = 1;
	required string name = 2;
	optional string did = 3;
	optional PlatForm platform = 4;

	optional string login_date = 5;
	optional string logout_date = 6;
	optional string reg_date = 7;

	optional uint32 vc1 = 8;
	optional uint32 vc2 = 9;
	optional uint32 vc3 = 10;

}